src/purchase-selection/utils.ts

```diff
@@ -0,0 +1,155 @@
+        0) ||
+    (product.limitations.appVersionMax &&
+      compareVersion(product.limitations.appVersionMax, input.appVersion) < 0)
+  )
+    return false;
+
+  if (
+    product.distributionChannel.some((channel) => channel === 'debug-app') &&
+    input.customerProfile?.debug
+  )
+    return true;
+
+  return product.distributionChannel.some((channel) => channel === 'app');
+};
+
+export const getDefaultProduct = (
+  input: PurchaseSelectionBuilderInput,
+  configType: string,
+) =>
+  input.preassignedFareProducts
+    .filter((product) => isProductSellableInApp(input, product))
+    .filter((product) => product.type === configType)
+    .reduce((selected, current) => (current.isDefault ? current : selected));
+
+export const getDefaultZone = (
+  input: PurchaseSelectionBuilderInput,
+  product: PreassignedFareProduct,
+): TariffZoneWithMetadata => {
+  const selectableZones = input.tariffZones.filter((zone) =>
+    isSelectableZone(product, zone),
+  );
+
+  if (input.currentCoordinates) {
+    const {longitude, latitude} = input.currentCoordinates;
+    const zoneFromLocation = selectableZones.find((t) =>
+      turfBooleanPointInPolygon([longitude, latitude], t.geometry),
+    );
+    if (zoneFromLocation) {
+      return {...zoneFromLocation, resultType: 'geolocation'};
+    }
+  }
+
+  const zone = selectableZones.reduce((selected, current) =>
+    current.isDefault ? current : selected,
+  );
+  return {...zone, resultType: 'zone'};
+};
+
+export const getDefaultUserProfiles = (
+  input: PurchaseSelectionBuilderInput,
+  product: PreassignedFareProduct,
+): UserProfileWithCount[] => {
+  const selectableProfiles = getSelectableUserProfiles(
+    input.userProfiles,
+    product,
+  );
+  const profile = selectableProfiles.reduce((selected, current) =>
+    current.userTypeString === input.defaultUserTypeString ? current : selected,
+  );
+  return [{...profile, count: 1}];
+};
+
+export const getSelectableUserProfiles = (
+  userProfiles: UserProfile[],
+  product: PreassignedFareProduct,
+): UserProfile[] =>
+  userProfiles.filter((profile) => isSelectableProfile(product, profile));
+
+export const isSelectableProduct = (
+  input: PurchaseSelectionBuilderInput,
+  selection: PurchaseSelectionType,
+  product: PreassignedFareProduct,
+) => {
+  if (selection.fareProductTypeConfig.type !== product.type) return false;
+  return isProductSellableInApp(input, product);
+};
+
+export const isSelectableProfile = (
+  product: PreassignedFareProduct,
+  profile: UserProfile,
+) => {
+  const profileLimitations = product.limitations.userProfileRefs;
+  return (
+    !profileLimitations.length ||
+    profileLimitations.some(
+      (allowedProfileId) => profile.id === allowedProfileId,
+    )
+  );
+};
+
+export const isSelectableZone = (
+  product: PreassignedFareProduct,
+  zone: TariffZone,
+) => {
+  const zoneLimitations = product.limitations.tariffZoneRefs;
+  return (
+    !zoneLimitations?.length ||
```

Comment:
This || here is a bit confusing especially because of a check with some.

Not sure if you want here is that product.limitations.tariffZoneRefs contains the zone, to be considered as selectable.

If !zoneLimitations?.length is true (meaning zoneLimitations is either absent or an empty array), the function immediately returns true without evaluating the some condition.

Is a selectable zone for an empty array? or undefined array?

Reply:
A zone is selectable if either there is no limitations (empty `zoneLimitations`) or if the limitations includes the given zone id.

I now created a boolean variable `emptyLimitations` that could make it a little more clear.

```ts
  const zoneLimitations = product.limitations.tariffZoneRefs;
  const emptyLimitations = !zoneLimitations?.length;
  return (
    emptyLimitations ||
    zoneLimitations.some((allowedZoneId) => zone.id === allowedZoneId)
  );
```